test/e2e/framework/util.go

```diff
@@ -3920,6 +3920,30 @@ func WaitForControllerManagerUp() error {
 	return fmt.Errorf("waiting for controller-manager timed out")
 }
 
+// CheckForControllerManagerHealthy checks that the controller manager does not crash within "duration"
+func CheckForControllerManagerHealthy(duration time.Duration) error {
+	var PID string
+	cmd := "sudo docker ps | grep k8s_kube-controller-manager | cut -d ' ' -f 1"
+	for start := time.Now(); time.Since(start) < duration; time.Sleep(5 * time.Second) {
+		result, err := SSH(cmd, GetMasterHost()+":22", TestContext.Provider)
+		if err != nil || result.Code != 0 {
+			// We don't necessarily know that it crashed, pipe could just be broken
+			Logf("kube-controller-manager unreachable at %v", time.Since(start))
+			LogSSHResult(result)
```

Comment:
If SSH always fails, then this function is going to return success.

Reply:
I guess a compromise must be made here. Maybe if the SSH fails a certain amount of times we call it a failure? But if it fails just once at some point we might not want to infer a controller-manager failure because it could just be networking issues. What do you think would be a reasonable way to approach this?

Reply:
Did you actually see the SSH fail during your testing?  I wouldn't expect it to.

Reply:
No I've never seen it fail. Just a thought. Should I just have it count as a failure?